src/Collection/CollectionTrait.php

```diff
@@ -987,10 +986,7 @@ public function isEmpty(): bool
     public function unwrap(): Iterator
     {
         $iterator = $this;
-        while (
-            $iterator::class === Collection::class
-            && $iterator instanceof OuterIterator
```

Comment:
This assumes only Collection uses CollectionTrait. We want to ensure that, but we're in a gray area now.